lib/streamlit/elements/altair.py

```diff
@@ -20,9 +20,25 @@
 
 
 import streamlit.elements.vega_lite as vega_lite
+import altair as alt
+import pandas as pd
 
 
-def marshall(vega_lite_chart, altair_chart, width=0):
+def generate_chart(chart_type, data):
+    if not isinstance(data, pd.DataFrame):
+        data = pd.DataFrame(data)
+
+    data = pd.melt(data.reset_index(), id_vars=['index'])
+
+    chart = getattr(alt.Chart(data), 'mark_' + chart_type)().encode(
+        alt.X('index', title=''),
+        alt.Y('value', title=''),
+        alt.Color('variable', title=''))
```

Comment:
Add `.interactive()` at the end

Also, make sure this chart shows tooltips when you hover over the datapoints.